### DISCUSSION TITLE
Allow `enabled` prop on `useSuspenseQuery`

### DISCUSSION THREAD

**Discussion by MrFlashAccount** [TanStack/query#7639] [General] [answered]
Hi! 

On our project, we have a bunch of `suspense` queries that depend on `env` parameters(if env flag is not provided, we skip this query), but this leads to a suboptimal experience  - fallback shows up for a moment. Simple repro can be found here: [Codesandbox](https://codesandbox.io/p/sandbox/hardcore-ardinghelli-qx45pp?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clxy8c2l500063b6mdkgzccrp%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clxy8c2l400023b6mg1s1j1i2%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clxy8c2l500033b6m21d76jzq%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clxy8c2l500053b6mepunezk4%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clxy8c2l400023b6mg1s1j1i2%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxy8c2l400013b6mkkxxj5iy%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%257D%255D%252C%2522id%2522%253A%2522clxy8c2l400023b6mg1s1j1i2%2522%252C%2522activeTabId%2522%253A%2522clxy8c2l400013b6mkkxxj5iy%2522%257D%252C%2522clxy8c2l500053b6mepunezk4%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxy8c2l500043b6msjqxfqk6%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clxy8c2l500053b6mepunezk4%2522%252C%2522activeTabId%2522%253A%2522clxy8c2l500043b6msjqxfqk6%2522%257D%252C%2522clxy8c2l500033b6m21d76jzq%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clxy8c2l500033b6m21d76jzq%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

I think it'd be a good idea to allow passing the `enabled` prop in `useSuspenseQuery`. In that case, `data` would have type `ReturnType<typeof queryFn> | InitalValue`

WDYT?

**Comment by hector** [comment]
Hey, I am facing a similar issue.
I have some suspense queries that I only need to really fetch the data depending on some parameter in the local storage.
I cannot conditionally call useSuspenseQuery because it is a hook. I believe there should be a parameter to bypass useSuspenseQuery so it does nothing.

**Comment by MrFlashAccount** [comment, marked as answer]
@hector this seems like nowadays the proper way is to use `useQuery` with `promise` and something like an `Await` component for react 18.
In react 19 its `use(promise)`

**Reply by hector** [reply]
Do you mean this?
https://tanstack.com/query/latest/docs/framework/react/guides/suspense#using-usequerypromise-and-reactuse-experimental

**Reply by MrFlashAccount** [reply]
Correct

**Reply by MrFlashAccount** [reply]
Also see: https://tanstack.com/router/v1/docs/framework/react/api/router/awaitComponent#:~:text=The%20Await%20component%20is%20a,the%20use()%20hook%20instead.
